### ISSUE TITLE
"FlowableException: Index: 1, Size: 1" on webservice invocation

### ISSUE THREAD

**Issue by cdeneux** [flowable/flowable-engine#2871]
The following error occurs at Apache CXF level when invoking a webservice returning 3 or more parameters. The problem does not occur if the webservice return 0, 1 or 2 parameters.

```
org.flowable.common.engine.api.FlowableException: Index: 1, Size: 1
	at org.flowable.engine.impl.webservice.CxfWebServiceClient.send(CxfWebServiceClient.java:84)
	at org.flowable.engine.impl.webservice.WSOperation.safeSend(WSOperation.java:83)
	at org.flowable.engine.impl.webservice.WSOperation.sendFor(WSOperation.java:63)
	at org.flowable.engine.impl.bpmn.webservice.Operation.sendMessage(Operation.java:54)
	at org.flowable.engine.impl.bpmn.behavior.WebServiceActivityBehavior.execute(WebServiceActivityBehavior.java:136)
        ...
Caused by: java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.ArrayList.rangeCheck(ArrayList.java:659)
	at java.util.ArrayList.get(ArrayList.java:435)
	at org.apache.cxf.jaxws.interceptors.HolderInInterceptor.handleMessage(HolderInInterceptor.java:67)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.endpoint.ClientImpl.onMessage(ClientImpl.java:826)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.handleResponseInternal(HTTPConduit.java:1693)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.handleResponse(HTTPConduit.java:1570)
	at org.apache.cxf.transport.http.HTTPConduit$WrappedOutputStream.close(HTTPConduit.java:1371)
	at org.apache.cxf.transport.AbstractConduit.close(AbstractConduit.java:56)
	at org.apache.cxf.transport.http.HTTPConduit.close(HTTPConduit.java:671)
	at org.apache.cxf.interceptor.MessageSenderInterceptor$MessageSenderEndingInterceptor.handleMessage(MessageSenderInterceptor.java:63)
	at org.apache.cxf.phase.PhaseInterceptorChain.doIntercept(PhaseInterceptorChain.java:308)
	at org.apache.cxf.endpoint.ClientImpl.doInvoke(ClientImpl.java:531)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:440)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:355)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:313)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:333)
	at org.apache.cxf.endpoint.ClientImpl.invoke(ClientImpl.java:319)
	at org.flowable.engine.impl.webservice.CxfWebServiceClient.send(CxfWebServiceClient.java:78)
	... 85 more
```

The CXF API `org.apache.cxf.endpoint.Client#invoke(methodName, arguments)` **must be called with `arguments` initialized also with expected output parameters as `null`** that is missing. For example for 2 input parameters and 3 output parameters: `client.invoke(methodName, new Object[] {inParam1, inParam2, null, null, null} )`.

This problem occurs on all versions of Flowable.

**Comment by cdeneux** [comment]
Hi Flowable team,

I'm working on a fix that I will propose you in few days.

**Comment by filiphr** [comment]
Closing this since it has been fixed in PR #2880, thanks @cdeneux
